src/passes/ConstantFieldPropagation.cpp

```diff
@@ -348,17 +380,25 @@ struct ConstantFieldPropagation : public Pass {
     //
     //  (struct.get $A x (REF-1))
     //
-    // Then we want to find all struct.sets
+    // Then we want to be aware of all the relevant struct.sets, that is, the
+    // sets that can write data that this get reads. Given a set
     //
     //  (struct.set $B x (REF-2) (..value..))
     //
-    // where $B is a subtype of $A, because at runtime the value REF-1 might not
-    // only be of type $A but also any subtype of $A. To make the lookup
-    // efficient when we see the get, we propagate information about sets to
-    // their supers, recursively, so long as the super still has that field.
-    // (Note that this also handles the case of REF-2 being a subtype of $B.)
+    // then
+    //
+    //  1. If $B is a subtype of $A, it is relevant: the actual value might be
+    //     of type $B (or even more specific), and cast to $A before it reaches
+    //     the get.
```

Comment:
```suggestion
    //     of type $B (or even more specific).
```
Even at the engine level, there's no cast when accessing a value that is dynamically a subtype of the statically accessed type.

Reply:
There's no cast in either case, so I think the mention of casts is part of what confused me at first.

Reply:
Yes, maybe "cast" is confusing. I meant that the value is passed into a location declared as a supertype (which is possible without casting).